front/lib/llm/providers/openai.ts

```diff
@@ -0,0 +1,163 @@
+import { OpenAI } from "openai";
+import type {
+  ResponseInputContent,
+  ResponseInputItem,
+} from "openai/resources/responses/responses.mjs";
+
+import { LLM } from "@app/lib/llm/llm";
+import type { LLMStreamEvent, ProviderMetadata } from "@app/lib/llm/types";
+import type {
+  Content,
+  RenderedModelConversationTypeMultiActions,
+} from "@app/types";
+
+export class OpenAILLM extends LLM {
```

Comment:
`OpenAIResponsesLLM` (will be used by xAI as well)

There will likely be a `OpenAICompletionLLM` for Deepseek and other